x/tx/signing/context.go

```diff
@@ -336,6 +340,8 @@ func (c *Context) getGetSignersFn(messageDescriptor protoreflect.MessageDescript
 	}
 	f, ok = c.getSignersFuncs[messageDescriptor.FullName()]
 	if !ok {
+		c.mtx.Lock()
+		defer c.mtx.Unlock()
 		var err error
 		f, err = c.makeGetSignersFunc(messageDescriptor)
```

Comment:
lgtm because `makeGetSignersFunc` is stateless right?  so worst case we get multiple blocked writes which will quickly resolve

Reply:
Yeap, before the worst case would bring the node down, now the worst case just creates some tiny delays